src/curve.ts

```diff
@@ -44,19 +71,21 @@ const curves = new Map([
   ["step-before", curveStepBefore]
 ]);
 
-export function Curve(curve = curveLinear, tension) {
+export function Curve(
+  curve: CurveName | CurveFunction = curveLinear,
+  tension: number
+): CurveFunction {
   if (typeof curve === "function") return curve; // custom curve
-  const c = curves.get(`${curve}`.toLowerCase());
+  const c = curves.get(`${curve}`.toLowerCase() as CurveName);
   if (!c) throw new Error(`unknown curve: ${curve}`);
```

Comment:
It works, but the logic escapes me a little: if _curve_ is not a curveName, the curves.get(…) must still be allowed to proceed.
For example `Plot.line(data, {curve: "BASIS"})` is correct, as is `Plot.line(data, {curve: {toString: () => "BASIS"}})`.

Reply:
👍 makes sense!

Reply:
spoke to Fil about this, we can have the types be case-sensitive but still let the js be more permissive, so as it's written is good